### PR TITLE
upkeep/163: Fix fatal error on PHP 5.6

### LINKED ISSUE #163: Running PHP 5.6 the plugin fatals before the PHP requirements check runs
### Describe the bug

The plugin includes a requirements check to ensure the PHP version installed is above PHP 7.4

Due to the return type definition on the function for the minimum supported version of PHP the plugin throws a fatal error on older versions of PHP

https://github.com/10up/simple-page-ordering/blob/06f69f62b6d2ab7df98dd5508a113d44476fee2f/simple-page-ordering.php#L32-L45

I suggest:

-  Removing the type hinting on the version check functions
-  Moving the main functionality of the plugin to another file
-  Only including the plugin functionality if the version check passes

### Steps to Reproduce

1.  Activate the plugin running WP 6.2 and PHP 7.4
2. Change the version of PHP to 5.6
3. Reload the WP dashboard
4. Observe the fatal error

### Screenshots, screen recording, code snippet

[19-Sep-2023 00:28:36 UTC] PHP Parse error:  syntax error, unexpected ':', expecting '{' in /vagrant/content/plugins/simple-page-ordering/simple-page-ordering.php on line 32
[19-Sep-2023 00:28:36 UTC] PHP Stack trace:
[19-Sep-2023 00:28:36 UTC] PHP   1. {main}() /vagrant/wp-build/wp-admin/plugins.php:0
[19-Sep-2023 00:28:36 UTC] PHP   2. require_once() /vagrant/wp-build/wp-admin/plugins.php:10
[19-Sep-2023 00:28:36 UTC] PHP   3. require_once() /vagrant/wp-build/wp-admin/admin.php:34
[19-Sep-2023 00:28:36 UTC] PHP   4. require_once() /vagrant/wp-build/wp-load.php:55
[19-Sep-2023 00:28:36 UTC] PHP   5. require_once() /vagrant/wp-config.php:146

### Environment information

_No response_

### WordPress information

WordPress 6.2
PHP 5.6

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

This PR does the following:

1. Moves the class `Simple_Page_Ordering` to a separate file.
2. Require `class-simple-page-ordering.php` only when platform validation passes.

Closes #163 

### How to test the Change

Please follow steps from the issue.

### Changelog Entry

> Fixed - Fatal errors on PHP 5.6

### Credits

Props @peterwilsoncc @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by iamdharmesh — CHANGES_REQUESTED** [review]
@Sidsector9 Thank you for the PR.

PR adds composer dependency to the plugin but it seems we have ignored the `vendor` folder in [.distignore](https://github.com/10up/simple-page-ordering/blob/develop/.distignore#L8) file. It may result in vendor folder missing in the release [zip](https://github.com/10up/simple-page-ordering/suites/16909797627/artifacts/966403674). Could you please help to check it once? 

Also, we have to add steps for composer package installation in our GH action workflows. especially in deploy workflow. Could you please help with it?

Thank you.

File: simple-page-ordering.php [hunk 1]
```diff
@@ -15,676 +15,19 @@
...
-							esc_html( minimum_php_requirement() )
-						)
-					);
-					?>
-				</p>
-			</div>
-			<?php
-		}
-	);
+if ( ! is_readable( __DIR__ . '/vendor/autoload.php' ) ) {
```
Comment (peterwilsoncc) [inline simple-page-ordering.php]: In the event further composer dependencies are added at a later date includingthis here may cause problems if the new dependencies require > PHP 5.6

Would it be possible to do one of the following:
* only require the compatibility checker in the vendor path
* install the compatibility checker outside of `vendor` and require it from the custom location

Comment (Sidsector9) [reply simple-page-ordering.php]: @peterwilsoncc I've taken the approach to move the library from `vendor` to `10up-lib` directory and also loaded the SPO main class after the validation is completed.

**Review by iamdharmesh — APPROVED** [review]
Thanks for the PR @Sidsector9. Code LGTM! and it tests well (Didn't throw any fatal error on 5.6 now).

In release build zip I have noticed some .md files in 10up-lib folder but it doesn't seems a blocker here. So, we are good to go here.
Screenshot 2023-10-09 at 10 38 19 PM

Thank you.

**Comment by Sidsector9** [comment]
Thanks for testing @iamdharmesh 
The files you're seeing is expected as you can also see them in the vendors directory. 
I'll resolve this in the next release of WP Compat Validator Tool.

**Pull request merged**